src/Key/AsymmetricKey.php

```diff
@@ -33,9 +34,9 @@ abstract class AsymmetricKey extends AbstractKey
     protected static function readFile(string $file): string
     {
         $key = file_get_contents($file);
-        if ($key === false) {
-            throw new InvalidArgumentException('Cannot read key from file "' . $file . '"');
-        }
+
+        Assert::true(is_string($key), 'Cannot read key from file "' . $file . '"', FileNotFoundException::class);
```

Comment:
The nitpick force is strong in me right now, so forgive me on beforehand, but there are multiple reasons why `file_get_contents()` may fail, not just because the file wasn't found 😜

Reply:
By the way, good job substituting all these checks by actual asserts! We didn't have them back in the day when I wrote this, and it's clearly paying off... 🥳

Reply:
I'm only replacing current code.. I'm not sure I can differentiate different kinds of `false` here..
So nitpick agreed upon, but maybe not for this PR unless you have an actual suggestion to improve this

Reply:
Yeah, I know. That's why I was thinking about a more generic exception... `IOException`? 🤔

Reply:
Strangely that's not a defined exception within the generic PHP exceptions... Not sure....  You want me to introduce the IOException and make FileNotFoundException extend from it?

Reply:
Related interested read: https://mike42.me/blog/2019-03-07-handling-io-errors-in-php

Reply:
Sounds like a good approach to me!